### PR TITLE
Add release notes for version 0.75.0, fix some typos

### PR DESCRIPTION
**Pull request by adamkvitek**
### Security

- [ ] Change has no security implications (otherwise, ping the security team)

### CODE REVIEW

File: docs/release-notes.md [hunk 1]
```diff
@@ -713,5 +769,5 @@ $ARG = [$V];
 
 ## Finding remaining release notes
 
-This document encompesses only a limit number of release notes for the purpose of clarity. To see all release notes, including older versions than displayed in this document, see [GitHub releases](https://github.com/returntocorp/semgrep/releases).
+This document encompasses only a limited number of release notes. See changelog that includes descriptions of older versions than displayed in this document: [Semgrep GitHub changelog](https://github.com/returntocorp/semgrep/releases)
```
Comment (pabloest) [inline docs/release-notes.md:772]:
```suggestion
This document encompasses only a limited number of release notes. See the changelog that includes descriptions of older versions than displayed in this document: [Semgrep GitHub changelog](https://github.com/returntocorp/semgrep/releases)
```

File: docs/release-notes.md [hunk 2]
```diff
@@ -24,7 +80,7 @@ With this release, Semgrep has improved the C++ parsing rate from 72.9% to 94.6%
 
 ##### Semgrep CI no longer fails scan with binary files
 
-Before this update, under certain circumstances, Semgrep reported `Pcre.Error(BadUTF8) error` when it tried to analyze PNG, TTF, EOT or WOFF, zip, tar and other binary files. As a consequence, scans failed when binary files were present. With this update, the underlying issue has been fixed and Semgrep skips binary files. ([#4258](https://github.com/returntocorp/semgrep/issues/4258))
+Before this update, under certain circumstances, Semgrep reported `Pcre.Error(BadUTF8) error` when it tried to analyze PNG, TTF, EOT or WOFF, zip, tar, and other binary files. As a consequence, scans failed when binary files were present. With this update, the underlying issue has been fixed, and Semgrep skips binary files. ([#4258](https://github.com/returntocorp/semgrep/issues/4258))
```
Comment (pabloest) [inline docs/release-notes.md:83]:
```suggestion
Before this update, Semgrep sometimes reported `Pcre.Error(BadUTF8) error` when it tried to analyze PNG, TTF, EOT or WOFF, zip, tar, and other binary files. As a consequence, scans failed when binary files were present. With this update, the underlying issue has been fixed, and Semgrep skips binary files. ([#4258](https://github.com/returntocorp/semgrep/issues/4258))
```

File: docs/release-notes.md [hunk 3]
```diff
@@ -12,7 +12,63 @@ Welcome to Semgrep release notes. This document provides an overview of the chan
...
 ## November 2021
 
-### Release 0.73.0
+### Version 0.75.0
+
+#### Fixes
+
+##### Semgrep CI
+
+In Semgrep CI, the `--disable-nosem` now tags findings with the `is_ignored` option correctly. Previously, an optimization from version 0.74.0 left the field `None` when the described option has been used. The optimization has been reverted.
```
Comment (pabloest) [inline docs/release-notes.md:21]:
```suggestion
In Semgrep CI, the option `--disable-nosem` now tags findings with the `is_ignored` option correctly. Previously, an optimization from version 0.74.0 left the field `None` when the described option has been used. The optimization has been reverted.
```

**Review by pabloest — COMMENTED** [review]
Added some small suggestions

**Review by s-santillan — COMMENTED** [review]
Looks good to me! @adamkvitek

**Review by s-santillan — APPROVED** [review]

**Pull request merged**